src/TreeHouse/BehatCommon/SecurityContext.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+namespace TreeHouse\BehatCommon;
+
+use Behat\Mink\Driver\BrowserKitDriver;
+use Behat\Mink\Exception\UnsupportedDriverActionException;
+use Behat\MinkExtension\Context\RawMinkContext;
+use Behat\Symfony2Extension\Context\KernelAwareContext;
+use PHPUnit_Framework_Assert as Assert;
+use Symfony\Component\BrowserKit\Cookie;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+/**
+ * @codeCoverageIgnore
+ */
+class SecurityContext extends RawMinkContext implements KernelAwareContext
+{
+    use KernelAwareTrait;
+
+    /**
+     * @var UserProviderInterface
+     */
+    private $userProvider;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var SessionInterface
+     */
+    private $symfonySession;
+
+    /**
+     * @var string
+     */
+    private $defaultLoginProperty;
+
+    /**
+     * @var string
+     */
+    private $defaultProviderKey;
+
+    /**
+     * @param UserProviderInterface $user_provider
+     * @param TokenStorageInterface $token_storage
+     * @param SessionInterface      $session
+     * @param string                $defaultLoginProperty
+     * @param string                $defaultProviderKey
+     */
+    public function __construct(
+        UserProviderInterface $user_provider,
+        TokenStorageInterface $token_storage = null,
+        SessionInterface $session = null,
+        $defaultLoginProperty = 'email',
```

Comment:
Dit zou username moeten zijn. Email zit standaard ook niet in de user interface.